api/tools/src/main/java/org/pmiops/workbench/tools/cdrconfig/UpdateCdrConfig.java

```diff
@@ -161,13 +175,19 @@ private void preCheck(CdrConfigVO cdrConfig) {
       }
     }
 
-    accessTierShortNames.forEach(
-        t -> {
-          if (!cdrDefaultVersionPerTier.containsKey(t)) {
-            throw new IllegalArgumentException(
-                String.format("Missing default CDR version for Access Tier '%s'.", t));
-          }
-        });
+    if (!allowEmptyTiers) {
+      accessTierShortNames.forEach(
+          t -> {
+            if (!cdrVersionsPerTier.containsKey(t)) {
+              throw new IllegalArgumentException(
+                  String.format("No CDR versions are present for Access Tier '%s'.", t));
+            }
+            if (!cdrDefaultVersionPerTier.containsKey(t)) {
```

Comment:
Even if we're allowing empty tiers, it seems like we should want to always make this assertion when the tier is non-empty

Reply:
good call.  Updated and tested.